src/main/java/com/github/princesslana/slothbot/Message.java

```diff
@@ -0,0 +1,39 @@
+package com.github.princesslana.slothbot;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.Set;
+
+public class Message {
```

Comment:
I go back and forth about this one. A Message class is definitely something that could exist, and I typically like just wrapping the JsonObject like is done here. But then I'm also thinking if it exists then it maybe shouldn't be tied to Gson in this project. If not this class, these methods could exist in the PermissionMiddleware or in the Discord class.